### PR TITLE
[FirmwareFlasher] Query DFU Device Inf Loop + Anykey Exit

### PR DESCRIPTION
**Pull request by krautech**
## Description

image

Loops until device is found OR any key is pressed

### CODE REVIEW

File: scripts/firmware.py [hunk 1]
```diff
@@ -1598,45 +1619,43 @@ def check_dfu_util(self) -> bool:
             print("dfu-util is not installed. Please install it and try again.")
             return False
 
-    def dfu_loop(self) -> List[str]:
-        start_time = time.time()
-        timeout = 30  # Timeout in seconds
+    def stop_loop(self, detected_devices: List[str]) -> List[str]:
+        return detected_devices
```
Comment (Jomik) [inline scripts/firmware.py:1623]: Why even have this? 😂

File: scripts/firmware.py [hunk 2]
```diff
@@ -1598,45 +1619,43 @@ def check_dfu_util(self) -> bool:
...
+                        _ = input("Press any key to return to the main menu.")
+                        return detected_devices  # Exit the loop and return devices
+
+                print(
+                    "DFU device not found, checking again... Press any key to return to the main menu."
+                )
+
+                # Check for key press with a timeout of 2 seconds
+                if Utils.is_key_pressed(timeout=2):
+                    return self.stop_loop(detected_devices)
```
Comment (Jomik) [inline scripts/firmware.py:1652]:
```suggestion
                    return detected_devices
```

File: scripts/firmware.py [hunk 3]
```diff
@@ -1598,45 +1619,43 @@ def check_dfu_util(self) -> bool:
...
+
         except KeyboardInterrupt:
-            print("\nQuery canceled by user.")
-            return []
+            return self.stop_loop(detected_devices)
```
Comment (Jomik) [inline scripts/firmware.py:1655]:
```suggestion
            return detected_devices
```

File: scripts/firmware.py (hunk 1 above)
Comment (krautech) [reply scripts/firmware.py:1623]: It was early in the AM :D think i had it a different way initially and just didnt remove it haha

**Review by Jomik — DISMISSED** [review]

**Review by Jomik — APPROVED** [review]

**Pull request merged**